aten/src/ATen/core/boxing/KernelFunction_impl.h

```diff
@@ -50,6 +62,13 @@ inline Return KernelFunction::call(const OperatorHandle& opHandle, Args... args)
     // forwarding, which would require Args to be deduced, but instead we
     // want callers to explicitly specify the Args.
 
+    TORCH_INTERNAL_ASSERT(!signature_hash_.has_value() || (detail::hashFunctionSignature<Return (Args...)>() == *signature_hash_),
```

Comment:
I've said this before and I'll say it again: I don't want an extra conditional test in the hotpath for all of our unboxed calls. If you bake the unboxed calling type into UnboxedHandle you can do the check once when we do the lookup (which is slow path anyway)

Reply:
fixed